skyportal/facility_apis/atlas.py

```diff
@@ -0,0 +1,310 @@
+59228.265532  17.223  0.049   469   23 c  0   3.90 342.45960  51.26340  2237.25  9167.94 4.31 3.88 -43.7 -0.473 18.95 21.20 02a59228o0152c
+                """
+
+                try:
+                    df = pd.read_csv(
+                        StringIO(s.text.replace("###", "")), delim_whitespace=True
+                    )
+                except Exception as e:
+                    raise ValueError(f'Format of response not understood: {e.message}')
+
+                desired_columns = ['MJD', 'RA', 'Dec', 'm', 'dm', 'mag5sig', 'F']
+                for col in desired_columns:
+                    if col not in list(set(df.columns.values)):
+                        raise ValueError(f'Missing expected column: {col}')
+
+                df.rename(
+                    columns={
+                        'MJD': 'mjd',
+                        'RA': 'ra',
+                        'Dec': 'dec',
+                        'm': 'mag',
+                        'dm': 'magerr',
+                        'mag5sig': 'limiting_mag',
+                        'F': 'filter',
+                    },
+                    inplace=True,
+                )
+                cyan = df['filter'] == 'c'
+                orange = df['filter'] == 'o'
+
+                snr = df['uJy'] / df['duJy'] < 5
+
+                df['filter'].loc[cyan] = 'atlasc'
+                df['filter'].loc[orange] = 'atlaso'
+                df['mag'].loc[snr] = None
+                df['magerr'].loc[snr] = None
+
+                iszero = df['duJy'] == 0.0
+                df['mag'].loc[iszero] = None
+                df['magerr'].loc[iszero] = None
+
+                isnan = np.isnan(df['uJy'])
+                df['mag'].loc[isnan] = None
+                df['magerr'].loc[isnan] = None
+
+                df = df.replace({np.nan: None})
+
+                drop_columns = list(
+                    set(df.columns.values)
+                    - set(
+                        ['mjd', 'ra', 'dec', 'mag', 'magerr', 'limiting_mag', 'filter']
+                    )
+                )
+
+                df.drop(
+                    columns=drop_columns,
+                    inplace=True,
+                )
+                df['magsys'] = 'ab'
+
+                data_out = {
+                    'obj_id': request.obj_id,
+                    'instrument_id': instrument.id,
+                    'group_ids': 'all',
+                    **df.to_dict(orient='list'),
+                }
+
+                from skyportal.handlers.api.photometry import PhotometryHandler
+
+                application = FakeApplication()
+                fake_request = FakeRequest()
+
+                photometry_handler = PhotometryHandler(
+                    application=application,
+                    request=fake_request,
+                )
+                photometry_handler.add_external_photometry(
```

Comment:
This is a risky move; did you check that error handling goes according to plan if PhotometryHandler fails?

Reply:
In https://github.com/skyportal/skyportal/pull/2648, all of these functions are completely pulled out. So It will just be a call to add_external_photometry.